server/src/main/java/keywhiz/KeywhizService.java

```diff
@@ -110,6 +110,15 @@ public void setInjector(Injector injector) {
 
   @SuppressWarnings("unchecked")
   @Override public void run(KeywhizConfig config, Environment environment) throws Exception {
+    try {
+      doRun(config, environment);
+    } catch (Exception e) {
+      logger.error(e.getMessage(), e);
+      throw e;
+    }
+  }
```

Comment:
After much experimentation, in the end we just needed to place this block in the correct position in order to get our exception stack trace into the logs:

```
ERROR [2021-08-05 23:19:23,827] keywhiz.KeywhizService: Validate failed: Migrations have failed validation
Detected resolved migration not applied to database: 14. To fix this error, either run migrate, or set -ignorePendingMigrations=true.
Need more flexibility with validation rules? Learn more: https://flywaydb.org/custom-validate-rules?ref=v7.11.3
! org.flywaydb.core.api.exception.FlywayValidateException: Validate failed: Migrations have failed validation
! Detected resolved migration not applied to database: 14. To fix this error, either run migrate, or set -ignorePendingMigrations=true.
! Need more flexibility with validation rules? Learn more: https://flywaydb.org/custom-validate-rules?ref=v7.11.3
! at org.flywaydb.core.Flyway$2.execute(Flyway.java:290)
! at org.flywaydb.core.Flyway$2.execute(Flyway.java:280)
! at org.flywaydb.core.Flyway.execute(Flyway.java:588)
! at org.flywaydb.core.Flyway.validate(Flyway.java:280)
! at keywhiz.KeywhizService.validateDatabase(KeywhizService.java:180)
! at keywhiz.KeywhizService.doRun(KeywhizService.java:170)
! at keywhiz.KeywhizService.run(KeywhizService.java:114)
! at keywhiz.KeywhizService.run(KeywhizService.java:72)
! at io.dropwizard.cli.EnvironmentCommand.run(EnvironmentCommand.java:43)
! at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:87)
! at io.dropwizard.cli.Cli.run(Cli.java:78)
! at io.dropwizard.Application.run(Application.java:93)
! at keywhiz.KeywhizService.main(KeywhizService.java:79)
```